### PR TITLE
Update player.lua

### PR DESCRIPTION
**Pull request by morpheause**
Fix for not being able to delete items in multiple slots & Function to get the total amount including items in multiple slots

### CODE REVIEW

**Comment by morpheause** [comment]
function name can also be getTotalQuantityByName

**Comment by Mojito-Fivem** [comment]
template not filled in

File: server/player.lua [hunk 1]
```diff
@@ -396,12 +396,26 @@ function QBCore.Player.CreatePlayer(PlayerData)
                         self.Functions.UpdatePlayerData()
                         TriggerEvent('qb-log:server:CreateLog', 'playerinventory', 'RemoveItem', 'red', '**' .. GetPlayerName(self.PlayerData.source) .. ' (citizenid: ' .. self.PlayerData.citizenid .. ' | id: ' .. self.PlayerData.source .. ')** lost item: [slot:' .. slot .. '], itemname: ' .. item .. ', removed amount: ' .. amount .. ', item removed')
                         return true
+                    elseif self.PlayerData.items[slot].amount < amountToRemove then
+                        amountToRemove = amountToRemove - self.PlayerData.items[slot].amount
+                        self.PlayerData.items[slot] = nil
+                        self.Functions.UpdatePlayerData()
+                        TriggerEvent('qb-log:server:CreateLog', 'playerinventory', 'RemoveItem', 'red', '**' .. GetPlayerName(self.PlayerData.source) .. ' (citizenid: ' .. self.PlayerData.citizenid .. ' | id: ' .. self.PlayerData.source .. ')** lost item: [slot:' .. slot .. '], itemname: ' .. item .. ', removed amount: ' .. amount .. ', item removed')
```
Comment (IdrisDose) [inline server/player.lua:403]: Remove this, please

**Review by IdrisDose — CHANGES_REQUESTED** [review]

**Comment by IllInuz** [comment]
Function seems to have been updated, can see some usage for it.

Though wouldn't it be easier to just do :

```lua
local itemdata = Player.Functions.GetItemByName('cash')
local amount = itemdata.amount
```
And then just return the amount?

**Comment by morpheause** [comment]
> Function seems to have been updated, can see some usage for it.
> 
> Though wouldn't it be easier to just do :
> 
> ```lua
> local itemdata = Player.Functions.GetItemByName('cash')
> local amount = itemdata.amount
> ```
> 
> And then just return the amount?

This returns the amount in the first slot it finds, not the total amount.

**Comment by morpheause** [comment]
> Function seems to have been updated, can see some usage for it.
> 
> Though wouldn't it be easier to just do :
> 
> ```lua
> local itemdata = Player.Functions.GetItemByName('cash')
> local amount = itemdata.amount
> ```
> 
> And then just return the amount?

This returns the amount in the first slot it finds, not the total amount.

File: server/player.lua (hunk 1 above)
Comment (morpheause) [reply server/player.lua]: image
image

Can you try this?

**Review by BerkieBb — CHANGES_REQUESTED** [review]
Edit, making a new one, wait

File: server/player.lua (hunk 1 above)
Comment (IdrisDose) [reply server/player.lua]: Why move all and return false? you're better off error-checking if you have the incorrect amount.

I see you did remove it though.

**Pull request closed without merging**